### PR TITLE
Using polymorphism to simplify object storage in canvas #16

### PR DESCRIPTION
**Pull request by alexeyleping**
https://github.com/1chooo/uml-editor/issues/16

### CODE REVIEW

**Comment by 1chooo** [comment]
Hi @alexeyleping 

Could you please attach a screenshot or a recording of this change to confirm that the app is still working correctly?

**Comment by 1chooo** [comment]
> Hi @alexeyleping
> 
> Could you please attach a screenshot or a recording of this change to confirm that the app is still working correctly?


I believe #16 and #17 will be included in one PR, which means #21 and #22 will be together in one PR. Please create **new PRs** that contain your excellent work! Could you write the commit messages in English? I'm sorry, but I cannot understand Russian.

**Comment by alexeyleping** [comment]
hi @1chooo
the sheet with line objects has been deleted, everything has been moved to the shapes sheet, and a check for class membership has been added
Screenshot from 2024-06-16 12-42-54

**Comment by 1chooo** [comment]
> hi @1chooo the sheet with line objects has been deleted, everything has been moved to the shapes sheet, and a check for class membership has been added Screenshot from 2024-06-16 12-42-54

So does it mean like we don't have to define `isInside()` in `BasicObject`, `Group`, `Line`. Because we define all in `Shape`. If yes, I don't think this is great choice which means if we want to draw more object, we cannot add the feature directly to the new graphic, we have to go back to `Shape`.

**Comment by 1chooo** [comment]
Please remove https://github.com/1chooo/uml-editor/pull/23/commits/34010279e4568b830ff7d86a2fa66a5a2555b2b4 and https://github.com/1chooo/uml-editor/pull/23/commits/cb70ce0e97a3bf9af2f7a5b20c713d3371f9d132 because they are not related to this PRs.

File: src/Utils/MODES.java [hunk 1]
```diff
@@ -1,30 +1,32 @@
 package Utils;
 
 public class MODES {
-    public static String SELECT = "Select";
-    public static String ASSOCIATION_LINE = "AssociationLine";
-    public static String GENERALIZATION_LINE = "GeneralizationLine";
-    public static String COMPOSITION_LINE = "CompositionLine";
-    public static String CLASS = "Class";
-    public static String USE_CASE = "Use Case";
+    public static final String SELECT = "Select";
```
Comment (1chooo) [inline src/Utils/MODES.java:4]: We have remove `final` in https://github.com/1chooo/uml-editor/pull/18

**Review by 1chooo — CHANGES_REQUESTED** [review]

**Pull request closed without merging**